.github/workflows/spherical_geometry.yml

```diff
@@ -0,0 +1,77 @@
+name: Spherical Geometry CI
+
+# Change 'on' section and add code coverage.
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "*"
+  pull_request:
+    branches:
+
+jobs:
+  build:
+    name: Python Testing ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9']
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -e .[test]
+    - name: Test with pytest
+      run: |
+        pytest
+
+  doc_build:
+    name: Document Build
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: |
+        sudo apt-get install graphviz texlive-latex-extra dvipng
+        python -m pip install --upgrade pip
+        python -m pip install -e .[docs]
+    - name: Build documents with sphinx
+      run: |
+        sphinx-build docs build/docs
+
+  code_coverage:
+    name: Code Coverage Report
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -e .[test]
+        python -m pip install pytest-cov
+    - name: Test with pytest and code coverage
+      run: |
+        pytest --cov-report=xml --cov=. --cov-config=setup.cfg
```

Comment:
Specifying that coverage config is in `setup.cfg` means you'll want to add the test dir to the omit section.  See `jwst` setup.cfg:

https://github.com/spacetelescope/jwst/blob/777015a87172b73d3d0ec58673712122907a57e0/setup.cfg#L115-L131

Reply:
I added an omit section to `setup.cfg`.